modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/DiagramHeader.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayButton from '@clayui/button';
+import React from 'react';
+
+const DiagramHeader = () => {
+	return (
+		<div className="d-flex diagram diagram-header justify-content-between">
+			<ClayButton
+				className="auto-mapping my-auto pull-right"
+				displayType="secondary"
+			>
+				Auto-mapping
+			</ClayButton>
+		</div>
+	);
+};
+
+DiagramHeader.defaultProps = {
```

Comment:
Why you declared defaultProps here since you're not using? Also, If you're using function components you can do something like:

```jsx
const INITIAL_RADIUS_CHOICE = [
		{
			label: 'Small',
			value: 10,
		},
		{
			label: 'Medium',
			value: 20,
		},
		{
			label: 'Large',
			value: 30,
		},
	];
	
const DiagramHeader = ({radiusChoice = INITIAL_RADIUS_CHOICE}) => {
```

Reply:
Sorry it's. refuse from the next 2 of 5 part of this component

Reply:
Yes, you can use just destructuring. Also, don't forget to use `Liferay.Language.get(labelValue)` for each label in the array :)